FDL: Add SwiftUI quickstart for link handling

andrewheard opened the pull request:
Added a SwiftUI-based quickstart for handling Firebase Dynamic Links.

paulb777 commented on dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift:
```diff
@@ -16,27 +16,58 @@ import SwiftUI
+    }
+  }
+
+  func handleLink(_ url: URL, completion: @escaping (DynamicLink?, Error?) -> Void) {
```
Can this be `async throws` instead?

andrewheard replied:
Good idea. This code has moved to [`extractDynamicLink`](https://github.com/firebase/quickstart-ios/pull/1428/files#diff-e25e2955aedb4999ed4a8c5d7f7bc26a9b474bfaaf1a0f92dc768e02ecc00100R101) but I made the change there.

paulb777 reviewed (commented):
Mostly looks good to me. 

Any updates needed for the instructions at https://github.com/firebase/quickstart-ios/blob/master/dynamiclinks/README.md beyond the iOs and Swift versions?

paulb777 commented on dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift:
```diff
@@ -43,8 +47,17 @@ struct DynamicLinksExampleApp: App {
+                  )
+                }
+              } catch UniversalURLError.notHandledByFDL {
+                print("FIRDynamicLinks is not handling the link.")
```
Don't use Objective-C "FIR" prefix here and below.

andrewheard replied:
This code no longer exists but noted for the future.

andrewheard commented:
> Mostly looks good to me.
> 
> Any updates needed for the instructions at https://github.com/firebase/quickstart-ios/blob/master/dynamiclinks/README.md beyond the iOs and Swift versions?

I updated the iOS version info and added a note about setting the `domainURIPrefix` in the SwiftUI example (same as the `DOMAIN_URI_PREFIX` constant in the previous ones).

morganchen12 reviewed (approved):
LGTM, please let me know if the snippets you added need to be in docs somewhere.

andrewheard commented:
> LGTM, please let me know if the snippets you added need to be in docs somewhere.

Thanks, Morgan. No need to update the snippets quite yet (just added them to prepare for FDL + SwiftUI setup docs in the future).
